opentelemetry-appender-log/CHANGELOG.md

```diff
@@ -2,6 +2,8 @@
 
 ## vNext
 
+## v0.3.0
```

Comment:
Given there is no breaking change here, do we need to use 0.3.0? A 0.2.1 would be more apt? 
@TommyCpp @hdost do you have thoughts on this? 

Or alternately, I'd prefer is all crates from this repo uses exact same version now, to make the whole releasing easier.
https://github.com/open-telemetry/opentelemetry-rust/pull/1555#discussion_r1493981609

Maybe stick the current practice for *this* release to unblock it quickly, and the next one onwards can adopt new versioning style, if we all agree.

Reply:
Technically the bump in OpenTelemetry would be a breaking change, no?

Reply:
And yea i think we mentioned in another thread about making all versions the same across components in the repo.

Reply:
So you can't really have both, right? Either you have semver-compatible versions in some crates or you have all the crates with the same version. Randomly skipping versions is possible but is also kinda weird for downstream consumers.

In the projects I'm involved with, we always stuck to proper semantic versioning. I think a decent compromise might be to document the version of the opentelemetry core crate that the crate is compatible with somewhere near the top of the README?

Reply:
yes we don't need to fix this part in this PR. Will use https://github.com/open-telemetry/opentelemetry-rust/pull/1555#discussion_r1493808966 to polish the process in the future.

Reply:
I'm trying to visualize this because what I am imagining is something like this. Semver only requires that you have x.y.z x is major, y minor and z patch. Since we're pre 1.0 any y can be breaking. But to my knowledge there's no requirement that for a given x,y,z that the number needs to be sequential. So going from v1 to v3 is technically acceptable if not "desirable". 

Here's what i am imagining for our purposes. Example is API/SDK represented with A and some dependent crate B. 

Today:
A - 0.21.0 -> 0.22.0
B - 0.2.1 -> 0.3.0

Same version example (Minor release)
A - 0.21.0 -> 0.22.0 -> 0.23.0
B - 0.2.1 -> 0.22.0 -> 0.23.0

Version skip 
(Same version) Patch
A - 0.21.0 -> 0.22.0 -> 0.22.1
B - 0.2.1 -> 0.22.0 -> nothing 

Next minor release 
A - 0.22.1 -> 0.23.0
B - 0.22.0 -> 0.23.0 

I can understand the initial confusion with the jump but that could be addressed in the changelog. 

And as I was mentioning before in the other thread I think it makes sense more in a pre-1.0 context because of how version resolution works.

Reply:
@hdost are you suggesting that we do it for *this* release onwards on the *next* one? I am okay with either. If we can start with *this* release, its better.

Reply:
> @hdost are you suggesting that we do it for *this* release onwards on the *next* one? I am okay with either. If we can start with *this* release, its better.

Also to clarify I don't think we should block this release on the conversation. For this one, let's keep the convention we're already doing for this one. We already have the other PR talking about release process.

Reply:
got it. Lets continue the current release with existing convention.

Reply:
BTW, opentelemetry-appender-log does have opentelemetry logs appearing in its public API, so it does actually need a semver-incompatible release number.